### DISCUSSION TITLE
What's the recomended backup code length?

### DISCUSSION THREAD

**Discussion by nskazki** [devise-two-factor/devise-two-factor#281] [Q&A]
Hello. Thank you for maintaining the gem that allowed me to effortlessly set up 2FA and have it working flawlessly for years!

In light of https://github.com/advisories/GHSA-qjxf-mc72-wjr2, bumping the default length of shared secrets from 120 to 128 bits, what is the recommended minimal length for backup codes? The one configured with `otp_backup_code_length`.

Regards, Nikita

**Comment by bsedat** [comment]
Hello! Just to be specific about the changes in v6.0.0 the new default length of the shared OTP secret is now 160 bits to match the recommended value in the spec.

I'm not aware of a recommended value in the spec for a backup code. The `otp_backup_code_length`'s new default is 16 random bytes, which then get encoded to a 32 byte hex string. That should have more possible combinations than the 6 character TOTP number. No matter what as discussed in [GHSA-chcr-x7hc-8fp8](https://github.com/devise-two-factor/devise-two-factor/security/advisories/GHSA-chcr-x7hc-8fp8) you'll need to make sure to prevent brute force attacks.

**Reply by nskazki** [reply]
Okay, thank you!
